server/apiui/writer.go

```diff
@@ -0,0 +1,74 @@
+package apiui
+
+import (
+	"strings"
+
+	"github.com/rancher/norman/api"
+	"github.com/rancher/norman/api/builtin"
+	"github.com/rancher/norman/api/writer"
+	"github.com/rancher/norman/types"
+	"github.com/rancher/rancher/pkg/settings"
+)
+
+var (
+	start = `
+<!DOCTYPE html>
+<!-- If you are reading this, there is a good chance you would prefer sending an
+"Accept: application/json" header and receiving actual JSON responses. -->
+<link rel="stylesheet" type="text/css" href="%API_UI_HOST_URL%/api-ui/%API_UI_VERSION_PATH%ui.min.css" />
+<script src="%API_UI_HOST_URL%/api-ui/%API_UI_VERSION_PATH%ui.min.js"></script>
```

Comment:
I assume you want to make this URL dynamic and norman isn't.  Can you just add field to the type in norman like
```golang
type URLGetter func() string

type HTMLResponseWriter struct {
	EncodingResponseWriter
        CSSURL URLGetter
        JSURL URLGetter
}
```
if CSSURL/JSURL is nil or returns "" then set to the const URL.  Then in rancher pass in a func that determines the URL dynamically.